app/src/main/java/husaynhakeem/io/facedetectorapp/MainActivity.kt

```diff
@@ -1,22 +1,39 @@
 package husaynhakeem.io.facedetectorapp
 
 import android.os.Bundle
+import android.util.Log
 import android.util.Size
 import androidx.appcompat.app.AppCompatActivity
 import com.otaliastudios.cameraview.Facing
+import husaynhakeem.io.facedetector.FaceBounds
 import husaynhakeem.io.facedetector.FaceDetector
 import husaynhakeem.io.facedetector.Frame
 import husaynhakeem.io.facedetector.LensFacing
 import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
 
+    private val onFaceDetectionResultListener = object : FaceDetector.OnFaceDetectionResultListener{
+        override fun onFailure(exception: Exception) {
+            super.onFailure(exception)
+            exception.printStackTrace()
+            Log.e(TAG, "error ${exception.message}")
+        }
+
+        override fun onSuccess(faceBounds: List<FaceBounds>) {
+            super.onSuccess(faceBounds)
+            Log.e(TAG,"total faces ${faceBounds.size}")
+            for (face in faceBounds) {
+                Log.d(TAG, "face ${face.id} ${face.box.width()} ${face.box.height()}")
+            }
+        }
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        val lensFacing =
-            savedInstanceState?.getSerializable(KEY_LENS_FACING) as Facing? ?: Facing.BACK
+        val lensFacing = Facing.FRONT
```

Comment:
`lensFacing` is saved in `onSaveInstanceState()`. Please revert setting it here to use the saved instance state.